### PR TITLE
Docs/#22 specifications

### PR DESCRIPTION
**Pull request by greenermoose**
Student coders have all had a chance to review and provide input into functional specifications.

### CODE REVIEW

File: Specs/Cataput.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+A cataput to shoot the balls into the high goal
+
+Needs a mathmatical model of the arm for better control
```
Comment (deslobodzian) [inline Specs/Cataput.txt:3]: Why is this? How we are currently planning on controlling doesn't need a dynamic model. All you need to know is how much string causes an exit angle of `x` and how much spring displacement causes a force `f`.

File: Specs/Indexer.txt [hunk 2]
```diff
@@ -0,0 +1,5 @@
+A method of moving the balls from the indexer to the cataput
+
+Needs to move the balls from the intake to the cataput quickly
+Needs a dejammer
+Needs a colour sensor to id the colour of the ball being indexed
```
Comment (deslobodzian) [inline Specs/Indexer.txt:5]: Indexer will not have a color sensor, the catapult will.

File: Specs/VSLAM.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Localize ourself on the field using a camera
+
+Needs relocalization
+Needs a method of denoising the recived data
+Needs to be accuate within a half millimeter
```
Comment (deslobodzian) [inline Specs/VSLAM.txt:5]: Extremely impractical.

File: Specs/VSLAM.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+Localize ourself on the field using a camera
+
+Needs relocalization
+Needs a method of denoising the recived data
```
Comment (deslobodzian) [inline Specs/VSLAM.txt:4]: Noise and drift are two different things.

File: docs/Specifications.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+## Will Do / Can Do Table
+
+As we write our specification and code we will fill out this table.
+
+Category | Specification | Will Do | Chassis Can Do | Comp Can Do
+---------|---------------|---------|----------------|------------
+Move | Move responding to remote human manual control | YES | NO | NO
+Move | Translate and rotate at the same time | YES | NO | NO
+Shoot | Shoot cargo into high goal | YES | NO | NO
+Shoot | Shoot and Move | NO | NO | NO
```
Comment (deslobodzian) [inline docs/Specifications.md:14]: Why can't we do this? If we have a vison system that gives us the velocity of the target, or localization that gives us the velocity components of the robot. This is mathematically easy.

File: docs/Specifications.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+Shoot | Check color using color sensor | YES | NO | NO 
+Shoot | Eject opponent cargo at low power| YES | NO | NO
+Auto | Complete an autonomous routine, scoring one to four cargo | YES | NO | NO
+Climb | Climb to Top Bar | YES | NO | NO
+V-SLAM | Localize Ourself on the field | YES | NO | NO
```
Comment (deslobodzian) [inline docs/Specifications.md:19]: If we can't do the advanced things like moving and shooting, what's the benefit of localization if our vision system works well?

File: Specs/Cataput.txt (hunk 1 above)
Comment (KilroyWasHere-cs-j) [reply Specs/Cataput.txt:3]: Will change.

File: Specs/VSLAM.txt (hunk 4 above)
Comment (KilroyWasHere-cs-j) [reply Specs/VSLAM.txt:4]: Yup, denoising doesn't necessarily fix drift.

File: docs/Specifications.md (hunk 5 above)
Comment (KilroyWasHere-cs-j) [reply docs/Specifications.md:14]: Noted

File: Specs/Indexer.txt (hunk 2 above)
Comment (KilroyWasHere-cs-j) [reply Specs/Indexer.txt:5]: Changed

**Comment by KilroyWasHere-cs-j** [comment]
Review again.

**Pull request closed without merging**